src/storage-controller/src/lib.rs

```diff
@@ -3259,4 +3267,39 @@ where
             }
         }
     }
+    /// Handles writing of status updates for sources/sinks to the appropriate
+    /// status relation
+    async fn record_status_updates(&mut self, updates: Vec<StatusUpdate<T>>) {
+        let mut sink_status_updates = vec![];
+        let mut source_status_updates = vec![];
+
+        for update in updates {
+            let id = update.id;
+            let update = healthcheck::RawStatusUpdate {
+                id,
+                // TODO(guswynn): Figure out a better way to extract the timestamp here. We know
+                // that currently `timestamp` must be a `u64`, but the compiler can't assume that it
+                // will always be. Since this timestamp always refers to the row representation,
+                // is there a need to make it generic?
+                ts: <u64 as Codec64>::decode(update.timestamp.encode()),
```

Comment:
I think this field should be a `chrono::DateTime` since that's what it ends up being and it's just the wall clock value that this status happened at

Reply:
this is SOOO much better omg